app/navigation/BusinessSettlorNavigator.scala

```diff
@@ -36,46 +37,70 @@ class BusinessSettlorNavigator @Inject()() extends Navigator {
+    case UkAddressPage | NonUkAddressPage => ua =>
+      navigateToEndPages(mode, trustType, ua)
+    case CompanyTimePage => ua =>
+      if (mode == NormalMode) {
+        rts.StartDateController.onPageLoad()
+      } else {
+        checkDetailsRoute(ua)
+      }
   }
 
-  private def navigationWithCheckAndTrustType(mode: Mode, trustType: TypeOfTrust): PartialFunction[Page, UserAnswers => Call] = {
-    mode match {
-      case NormalMode => {
-        case UtrPage | UkAddressPage | NonUkAddressPage => _ =>
-          trustType match {
-            case EmployeeRelated =>
-              rts.CompanyTypeController.onPageLoad(mode)
-            case _ =>
-              rts.StartDateController.onPageLoad()
-          }
-        case CompanyTimePage => _ =>
-          rts.StartDateController.onPageLoad()
-        case AddressYesNoPage => ua =>
-          yesNoNav(ua, AddressYesNoPage, rts.LiveInTheUkYesNoController.onPageLoad(mode), rts.StartDateController.onPageLoad())
-      }
-      case CheckMode => {
-        case UtrPage | UkAddressPage | NonUkAddressPage => ua =>
-          trustType match {
-            case EmployeeRelated =>
-              rts.CompanyTypeController.onPageLoad(mode)
-            case _ =>
-              checkDetailsRoute(ua)
-          }
-        case CompanyTimePage => ua =>
-          checkDetailsRoute(ua)
-        case AddressYesNoPage => ua =>
-          yesNoNav(ua, AddressYesNoPage, rts.LiveInTheUkYesNoController.onPageLoad(mode), checkDetailsRoute(ua))
-      }
+
+  private def navigateAwayFromNamePage(mode: Mode, answers: UserAnswers): Call = {
+    if (answers.is5mldEnabled && !answers.isTaxable) {
+      rts.CountryOfResidenceYesNoController.onPageLoad(mode)
+    } else {
+      rts.UtrYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateAwayFromUtrPages(mode: Mode, trustType: TypeOfTrust, answers: UserAnswers): Call = {
+    (answers.is5mldEnabled, answers.get(UtrYesNoPage)) match {
+      case (true, _) => rts.CountryOfResidenceYesNoController.onPageLoad(mode)
+      case (false, Some(true)) => navigateToEndPages(mode, trustType, answers)
+      case (false, _) => rts.AddressYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateAwayFromResidencePages(mode: Mode, trustType: TypeOfTrust, answers: UserAnswers): Call = {
+    val hasUtr = answers.get(UtrYesNoPage).getOrElse(false)
+    val isNonTaxable5mld = answers.is5mldEnabled && !answers.isTaxable
+
+    if (isNonTaxable5mld || hasUtr){
+      navigateToEndPages(mode, trustType, answers)
+    } else {
+     rts.AddressYesNoController.onPageLoad(mode)
+    }
+  }
+
+  private def navigateToEndPages(mode:Mode, trustType: TypeOfTrust, ua: UserAnswers): Call = {
+    (mode, trustType) match {
+      case (_, EmployeeRelated) => rts.CompanyTypeController.onPageLoad(mode)
+      case (NormalMode, _) => rts.StartDateController.onPageLoad()
+      case (CheckMode, _) => checkDetailsRoute(ua)
```

Comment:
I would consider extracting lines 64-68 into it's own method then here do if trustType is EmployeeRelated go to CompanyType else call that other method

Reply:
Sure. Done